commandbasedarchitecture_coroutine/src/main/java/io/scal/commandbasedarchitecture/broadcast/BaseBroadcastCommandViewModel.kt

```diff
@@ -1,53 +1,94 @@
 package io.scal.commandbasedarchitecture.broadcast
 
-import android.util.Log
+import android.os.Parcel
+import android.os.Parcelable
 import androidx.lifecycle.LiveData
 import androidx.lifecycle.MutableLiveData
 import io.scal.commandbasedarchitecture.ActionCommand
 import io.scal.commandbasedarchitecture.CommandManager
 import io.scal.commandbasedarchitecture.CommandManagerImpl
 import io.scal.commandbasedarchitecture.DataConvertCommandSameResult
+import io.scal.commandbasedarchitecture.model.readNullOrValue
+import io.scal.commandbasedarchitecture.model.writeToParcel
+import kotlinx.android.parcel.Parceler
+import kotlinx.android.parcel.Parcelize
 import kotlinx.coroutines.CoroutineScope
 import kotlinx.coroutines.Dispatchers
 import java.lang.ref.WeakReference
 
-interface ChildViewModel<ChildKey> {
-    val key: ChildKey
+interface ChildViewModel<ChildState> {
+    val fullDataState: LiveData<ChildState>
 }
 
 /**
  * Broadcast data state class
  *
- * @param hardViewModels view models that should be kept permanently after the first request
- * @param weakViewModels view models that will be kept until cleared by gc
+ * @param hardViewStates view models that should be kept permanently after the first request
```

Comment:
IMO, `strongViewStates` is likely a better name for this, since it's a more standard software nomenclature.

Reply:
done!